src/Raven.Server/Rachis/RachisConsensus.cs

```diff
@@ -126,39 +127,40 @@ public void Dispose()
             }
         }
 
-        public unsafe List<BlittableJsonReaderObject> GetLogEntries<TTransaction>(long first, TransactionOperationContext<TTransaction> context, int max)
-            where TTransaction : RavenTransaction
+        public IEnumerable<BlittableJsonReaderObject> GetLogEntries(ClusterOperationContext context, long first, int start, int take)
         {
-            var entries = new List<BlittableJsonReaderObject>();
             var reveredNextIndex = Bits.SwapBytes(first);
+            Span<byte> span = stackalloc byte[sizeof(long)];
+            if (BitConverter.TryWriteBytes(span, reveredNextIndex) == false)
+                throw new InvalidOperationException($"Couldn't convert {first} to span<byte>");
+
             var table = context.Transaction.InnerTransaction.OpenTable(LogsTable, EntriesSlice);
-            using (Slice.External(context.Allocator, (byte*)&reveredNextIndex, sizeof(long), out Slice key))
+            using (Slice.From(context.Allocator, span, out Slice key))
             {
-                foreach (var value in table.SeekByPrimaryKey(key, 0))
+                foreach (var value in table.SeekByPrimaryKey(key, start))
                 {
-                    var entry = FollowerAmbassador.BuildRachisEntryToSend(context, value);
-                    Transaction.DebugDisposeReaderAfterTransaction(context.Transaction.InnerTransaction, entry);
-                    entries.Add(entry);
-                    if (entries.Count >= max)
-                        break;
+                    if (take-- <= 0)
+                        yield break;
+
+                    yield return FollowerAmbassador.BuildRachisEntryToSend(context, value);
                 }
             }
-            return entries;
         }
     }
 
     public class RachisLogEntry : IDynamicJsonValueConvertible
     {
         public DateTime At = DateTime.UtcNow;
         public string Message;
-        public long Ticks;
+        public long MsFromCycleStart;
```

Comment:
is this in-memory state only or sth that we are persisting?

Reply:
in memory only